src/wp-includes/rest-api/endpoints/class-wp-rest-font-library-controller.php

```diff
@@ -0,0 +1,394 @@
+			$font_family = $font_families[ $family_index ];
+
+			if (
+				! isset( $font_family['slug'] ) ||
+				! isset( $font_family['name'] ) ||
+				! isset( $font_family['fontFamily'] )
+			) {
+				$error_messages[] = sprintf(
+					// translators: 1: font family index.
+					__( 'Font family [%s] should have slug, name and fontFamily properties defined.' ),
+					$family_index
+				);
+			}
+
+			if ( isset( $font_family['fontFace'] ) ) {
+				if ( ! is_array( $font_family['fontFace'] ) ) {
+					$error_messages[] = sprintf(
+						// translators: 1: font family index.
+						__( 'Font family [%s] should have fontFace property defined as an array.' ),
+						$family_index
+					);
+					continue;
+				}
+
+				if ( count( $font_family['fontFace'] ) < 1 ) {
+					$error_messages[] = sprintf(
+						// translators: 1: font family index.
+						__( 'Font family [%s] should have at least one font face definition.' ),
+						$family_index
+					);
+				}
+
+				if ( ! empty( $font_family['fontFace'] ) ) {
+					for ( $face_index = 0; $face_index < count( $font_family['fontFace'] ); $face_index++ ) {
+
+						$font_face = $font_family['fontFace'][ $face_index ];
+						if ( ! isset( $font_face['fontWeight'] ) || ! isset( $font_face['fontStyle'] ) ) {
+							$error_messages[] = sprintf(
+								// translators: 1: font family index, 2: font face index.
+								__( 'Font family [%1$s] Font face [%2$s] should have fontWeight and fontStyle properties defined.' ),
+								$family_index,
+								$face_index
+							);
+						}
+
+						if ( isset( $font_face['downloadFromUrl'] ) && isset( $font_face['uploadedFile'] ) ) {
+							$error_messages[] = sprintf(
+								// translators: 1: font family index, 2: font face index.
+								__( 'Font family [%1$s] Font face [%2$s] should have only one of the downloadFromUrl or uploadedFile properties defined and not both.' ),
+								$family_index,
+								$face_index
+							);
+						}
+
+						if ( isset( $font_face['uploadedFile'] ) ) {
+							if ( ! isset( $files[ $font_face['uploadedFile'] ] ) ) {
+								$error_messages[] = sprintf(
+									// translators: 1: font family index, 2: font face index.
+									__( 'Font family [%1$s] Font face [%2$s] file is not defined in the request files.' ),
+									$family_index,
+									$face_index
+								);
+							}
+						}
+					}
+				}
+			}
+		}
+
+		return $error_messages;
+	}
+
+	/**
+	 * Validate input for the install endpoint.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string          $param The font families to install.
+	 * @param WP_REST_Request $request The request object.
+	 * @return true|WP_Error True if the parameter is valid, WP_Error otherwise.
+	 */
+	public function validate_install_font_families( $param, $request ) {
+		$font_families  = json_decode( $param, true );
+		$files          = $request->get_file_params();
+		$error_messages = $this->get_validation_errors( $font_families, $files );
```

Comment:
It would be cleaner if this returns a WP_errror object.

Reply:
We are returning the WP_Error a few lines below: https://github.com/WordPress/wordpress-develop/blob/2e08b5374fbed678ca2328a52234a6ce972f0353/src/wp-includes/rest-api/endpoints/class-wp-rest-font-library-controller.php#L247

Reply:
@matiasbenedetto This is what I had in mind. https://github.com/WordPress/wordpress-develop/commit/8e4a9a9182b0726beda37a124f4a92bf596467a7

This would also make it translations easier as well.